articles/choosing-between-firebase-and-mongodb/index.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: engineering-education
+status: publish
+published: true
+slug: choosing-between-firebase-and-mongo-db
+title: Choosing Between Firebase and MongoDB
+description: Respective history and pros and cons of using Firebase or MongoDB in a tech stack.
+date: 2020-11-09T00:00:00-07:00
+topics: [databases][mongodb][firebase]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/choosing-between-firebase-and-mongo-db/hero.jpg
+    alt: Relational Databases
+---
+
+### MongoDB vs Firebase
+Databases play an important role in determining the functionality and quality of an application or a website. For an app to be successful,
+the developer will have to consider the users of the application or the website.
+The quantity of data stored and retrieved from the application is one of the factors considered when choosing a database.
+Other factors considered when choosing the right database for an application or a website are:
+1. Latency.
+Latency refers to the response time of the database and to the end-to-end response time of the application. 
+2.structure of the data. Choosing the right database for an application or a website involves the developer understanding the right data structures for storage and retrieval of data. Various applications deal with data of various formats In this article, I am going to focus on Firebase and MongoDB while highlighting its differences.
+### History of MongoDB
+10gen company which is now MongoDB Inc developed MongoDB in the year 2007. The 10gen main language was server-side JavaScript.10gen company started to develop as a Platform as a service ( PAAS) product, but they experienced scalability issues with the existing relational database systems. They developed a document-oriented database system called MongoDB.
+### Introduction to MongoDB
+MongoDB is an open-source document-oriented database that stores data in JSON-like formats. MongoDB makes use of documents and collections instead of using tables and rows. It supports the dynamic schema design. MongoDB is a document-oriented server developed in a C++ programming language.
+### Features of MongoDB
+1. Native Replication.
+MongoDB provides a replication feature by distributing data across different machines through a replica set topology.According to [Congruent Solutions](https://www.congruentsolutions.com/blogposts/mongodb-replication-approach-setup-using-arbiter/),A replica set is a group of MongoDB instances that host the same data set.In a replica set one node is primary node that receives all write operations and all other nodes acts as secondaries. Replica set can have only one primary node.After each successive writes to the primary node, MongoDB will replicate the same data to all the secondary nodes present in the replica set.
```

Comment:
Replica set topology - break down

Spacing -> "Accordingly..."

Odd capitalization + spacing -> ",A replica"

Spacing throughout??? There are no spaces after periods.

In a replica set one node-> run on, needs comma

"In a replica set one node is primary node that receives all write operations and all other nodes acts as secondaries." => what does this mean??

"Replica set can have only one primary node." -> GRAMMAR. 

The concept of why there is only one primary node is confusing - simplify.